Makefile.build

```diff
@@ -1,389 +0,0 @@
-##########################################################################
-##         #   The Coq Proof Assistant / The Coq Development Team       ##
-##  v      #         Copyright INRIA, CNRS and contributors             ##
-## <O___,, # (see version control and CREDITS file for authors & dates) ##
-##   \VV/  ###############################################################
-##    //   #    This file is distributed under the terms of the         ##
-##         #     GNU Lesser General Public License Version 2.1          ##
-##         #     (see LICENSE file for the text of the license)         ##
-##########################################################################
-
-# This makefile is normally called by the main Makefile after setting
-# some variables.
-
-###########################################################################
-# User-customizable variables
-###########################################################################
-
-# The following variables could be modified via the command-line of make,
-# either with the syntax 'make XYZ=1' or 'XYZ=1 make'
-
-# To see the actual commands launched by make instead of shortened versions,
-# set this variable to 1 (or any non-empty value):
-VERBOSE ?=
-
-# When non-empty, a time command is performed at each .v compilation.
-# To collect compilation timings of .v and import them in a spreadsheet,
-# you could hence consider: make TIMED=1 2> timings.csv
-TIMED ?=
-
-# When $(TIMED) is set, the time command used by default is $(STDTIME)
-# (see below), unless the following variable is non-empty. For instance,
-# it could be set to "'/usr/bin/env time -p'".
-TIMECMD ?=
-
-# When non-empty, -time is passed to coqc and the output is recorded
-# in a timing file for each .v file.  If set to "before" or "after",
-# the file name for foo.v is foo.v.$(TIMING)-timing; otherwise, it is
-# foo.v.timing
-TIMING ?=
-
-# Non-empty runs the checker on all produced .vo files:
-VALIDATE ?=
-
-# When non-empty, passed as extra arguments to coqtop/coqc:
-COQUSERFLAGS ?=
-
-# Option for changing sorting of timing output file
-TIMING_SORT_BY ?= auto
-# Option for changing the fuzz parameter on the output file
-TIMING_FUZZ ?= 0
-# Option for changing whether to use real or user time for timing tables
-TIMING_REAL?=
-# Option for including the memory column(s)
-TIMING_INCLUDE_MEM?=
-# Option for sorting by the memory column
-TIMING_SORT_BY_MEM?=
-# Output file names for timed builds
-TIME_OF_BUILD_FILE               ?= time-of-build.log
-TIME_OF_BUILD_BEFORE_FILE        ?= time-of-build-before.log
-TIME_OF_BUILD_AFTER_FILE         ?= time-of-build-after.log
-TIME_OF_PRETTY_BUILD_FILE        ?= time-of-build-pretty.log
-TIME_OF_PRETTY_BOTH_BUILD_FILE   ?= time-of-build-both.log
-TIME_OF_PRETTY_BUILD_EXTRA_FILES ?= - # also output to the command line
-BEFORE ?=
```

Comment:
Do we time builds now?

Reply:
Actually it would be great to get rid of the regexp based timing system we have and use something better, because the current one is a big hack.

The code that I'm removing here just concerns to the stdlib, and was (poorly duplicated) from coq_makefile, where it is still present. We can extend gen_rules to produce a similar timing log if there is demand. Not sure I'd bother, use case is really small, I'd rather implement #11542

Reply:
This time-of-build stuff isn't used in benching AFAIK

Reply:
We should def bench this, but I removed this indeed under the assumption Gaëtan points out, we are not really using it, and it is actually not very good maintained because we only test the bits that are in coq_makefile, which of course are legit and not scheduled for removal in any way.